### PR TITLE
Prevent unnecessary CLI calls in the integration test suite

### PR DESCRIPTION
**Pull request by mattseddon**
After seeing #2929 fail numerous times on 🪟s I went down the rabbit hole of trying to completely clean our integration test logs (and thus clean up unexpected behaviour). Found some very weird behaviour and dead code but think I got to the bottom of it all. Comments inline.

### CODE REVIEW

File: extension/src/experiments/data/index.ts [hunk 1]
```diff
@@ -66,10 +66,6 @@ export class ExperimentsData extends BaseData<ExperimentsOutput> {
     return this.notifyChanged(data)
   }
 
-  public forceUpdate() {
```
Comment (mattseddon) [inline extension/src/experiments/data/index.ts:69]: [F] All the force update stuff is was dead.

File: extension/src/processManager.ts [hunk 2]
```diff
@@ -55,17 +55,6 @@ export class ProcessManager extends Disposable {
     return this.runQueued()
   }
 
-  public async forceRunQueued(): Promise<void> {
```
Comment (mattseddon) [inline extension/src/processManager.ts:58]: [F] Only place this was called was a test. I was in this class in the first place to try and stop the queue from being run after it had been disposed of.

File: extension/src/workspace/index.ts [hunk 3]
```diff
@@ -38,7 +39,9 @@ export abstract class BaseWorkspace<
   }
 
   public reset(): void {
-    this.repositories = reset<T>(this.repositories, this.dispose)
```
Comment (mattseddon) [inline extension/src/workspace/index.ts:41]: [F] I wanted to be sure that `this.dispose` was not unbounded

File: extension/src/test/suite/extension.test.ts [hunk 4]
```diff
@@ -447,6 +449,7 @@ suite('Extension Test Suite', () => {
     })
 
     it('should set the dvc.cli.incompatible context value', async () => {
+      stub(Watcher, 'createFileSystemWatcher').returns(mockDisposable)
```
Comment (mattseddon) [inline extension/src/test/suite/extension.test.ts:452]: [F] Don't want to accidentally setup all the watchers here.

**Review by sroy3 — APPROVED** [review]

**Review by julieg18 — APPROVED** [review]
Looks good!

**Pull request merged**